src/analysis/fourier/poisson_summation.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import analysis.fourier.add_circle
+import analysis.fourier.fourier_transform
+import measure_theory.integral.exp_decay
+import analysis.special_functions.gaussian
+
+/-!
+# Poisson's summation formula
+
+We prove Poisson's summation formula `∑ (n : ℤ), f n = ∑ (n : ℤ), 𝓕 f n`, where `𝓕 f` is the
+Fourier transform of `f`, under the following hypotheses:
+* `f` is a continuous function `ℝ → ℂ`.
+* The sum `∑ (n : ℤ), 𝓕 f n` is convergent.
+* For all compacts `K ⊂ ℝ`, the sum `∑ (n : ℤ), sup { ‖f(x + n)‖ | x ∈ K }` is convergent.
+
+The last condition is automatic if we assume `f` has sufficiently rapid decay (e.g. if it is a
+Schwartz function); this will be added in future.
+-/
+
+noncomputable theory
+
+open function (hiding comp_apply) complex real set topological_space filter measure_theory
+
+open_locale real big_operators filter fourier_transform
+
+local attribute [instance] real.fact_zero_lt_one
+
+namespace continuous_map
+
+variables
+  {α : Type*} [topological_space α]
+  {E : Type*} [normed_add_comm_group E]
+
+/-- The continuous map `λ x, x * m`. -/
+@[to_additive "The continuous map `λ x, x + m`", simps]
+def mul_right [has_mul α] [has_continuous_mul α] (m : α) : C(α, α) := mk _ (continuous_mul_right m)
```

Comment:
This already exists as `homeomorph.mul_right`.

Reply:
The existing declaration is a slightly different type -- a homeomorph, not a continuous map -- and accordingly it is defined only for topological groups, not just topological monoids (where `mul_right` is a well-defined continuous map but might not be a homeomorph). 

This declaration has gone through several iterations now (cf. Eric's comment [here](https://github.com/leanprover-community/mathlib/pull/18392#discussion_r1098768061)), and each iteration has had the side-effect of making the main theorems of the file messier to state -- what was initially `∑' n : ℤ, f.comp_add_right n` became `∑' n : ℤ, f.comp (continuous_map.add_right n)` and would now have to become `∑' n : ℤ, f.comp (homoemomorph.add_right n).to_continuous_map`. 

How about get rid of the `continuous_map.mul_right` declaration, but reinstating something closer to the first iteration? I'm thinking of something like
```
@[to_additive, simps]
def continuous_map.comp_mul_right [some typeclasses] (f : C(α, β)) (a : α) : C(α, β) := 
f.comp (homoemomorph.mul_right a).to_continuous_map
``` 
Then we get the dot-notation `f.comp_add_right a` back again, and we also get the an autogenerated `comp_mul_right_apply` simplification lemma, which will shorten some of the arguments in the file considerably. What do you think?

Reply:
That's a good point. Then you can keep it as is and move it to `topology.algebra.monoid`. `continuous_map.comp_mul_right` looks strictly worse, and I wouldn't recommend having it, except possibly as a private reducible def.

Reply:
@YaelDillies, I can't quite work out the meaning of your comment above. Are you against having `continuous_map.mul_right`?

Reply:
Let us discuss this further at the spinoff PR #18465.